services:
  nginx:
    image: nginx:1.27.4-alpine
    volumes:
      - ./nginx/default.conf:/etc/nginx/conf.d/default.conf
      - ./certbot/conf:/etc/letsencrypt
      - ./certbot/www:/var/www/certbot
    ports:
      - "80:80"
      - "443:443"
    environment:
      - NGINX_HOST=api.rikko.space
    restart: unless-stopped
    depends_on:
      - server
    networks:
      - nginx-network

  database:
    image: postgres
    ports:
      - "5432:5432"
    environment:
      - PGUSER=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_DB=auth
      - POSTGRES_PASSWORD=postgres
    volumes:
      - store_database_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - database_network

  cache:
    image: redis:8.0-rc1-alpine
    restart: always
    ports:
      - '6379:6379'
    command: redis-server --save 20 1 --loglevel warning --requirepass eYVX7EwVmmxKPCDmwMtyKVge8oLd2t81
    volumes:
      - cache:/data

  server:
    build: ./Server
    container_name: auth_server_c
    ports:
      - "8080:8080"
    env_file:
      - .env
    volumes:
      - secrets:/app/secrets
    depends_on:
      - database
      - cache
    restart: unless-stopped
    networks:
      - nginx-network
      - database_network
        
  migration:
    build: ./MigrationUtility
    container_name: migration_utility_c
    env_file:
      - .env
    depends_on:
      - database
    restart: no
    networks:
      - database_network

  webapp:
    build:
      context: ./website
      dockerfile: Dockerfile
    environment:
      - NODE_ENV=production
    networks:
      - nginx-network
    # depends_on:
    #  - server



volumes:
  store_database_data:
  secrets:
  cache:
      driver: local
    
networks:
  nginx-network:
    driver: bridge
  database_network:
    driver: bridge




